sdk/identity/azure-identity/Troubleshoot.md

```diff
@@ -0,0 +1,223 @@
+# Troubleshooting Azure Identity Authentication Issues
+
+The Azure Identity SDK offers various `TokenCredential` implementations. These implementations typically throw `CredentialUnavailableError` and `ClientAuthenticationError` exceptions.
+The `CredentialUnavailableError` indicates that the credential cannot execute in the current environment setup due to lack of required configuration.
+The `ClientAuthenticationError` indicates that the credential was able to run/execute but ran into an authentication issue from the server's end. This can happen due to invalid configuration/details passed in to the credential at construction time.
+This troubleshooting guide covers mitigation steps to resolve these exceptions thrown by various `TokenCredential` implementations in the Azure Identity Python client library.
+
+## Table of contents
+
+- [Troubleshooting Default Azure Credential Authentication Issues](#troubleshooting-default-azure-credential-authentication-issues)
+- [Troubleshooting Environment Credential Authentication Issues](#troubleshooting-environment-credential-authentication-issues)
+- [Troubleshooting Service Principal Authentication Issues](#troubleshooting-service-principal-authentication-issues)
+- [Troubleshooting Username Password Authentication Issues](#troubleshooting-username-password-authentication-issues)
+- [Troubleshooting Managed Identity Authentication Issues](#troubleshooting-managed-identity-authentication-issues)
+- [Troubleshooting Visual Studio Code Authentication Issues](#troubleshooting-visual-studio-code-authentication-issues)
+- [Troubleshooting Azure CLI Authentication Issues](#troubleshooting-azure-cli-authentication-issues)
+- [Troubleshooting Azure Powershell Authentication Issues](#troubleshooting-azure-powershell-authentication-issues)
+
+## Troubleshooting Default Azure Credential Authentication Issues
+
+### Credential Unavailable Error
+
+The `DefaultAzureCredential` attempts to retrieve an access token by sequentially invoking a chain of credentials. The `CredentialUnavailableError` in this scenario signifies that all the credentials in the chain failed to retrieve the token in the current environment setup/configuration. You need to follow the configuration instructions for the respective credential you're looking to use via `DefaultAzureCredential` chain, so that the credential can work in your environment.
```

Comment:
`DefaultAzureCredential` doesn't raise `CredentialUnavailableError` (in any case). If all its constituent credentials do so, `DefaultAzureCredential` raises `ClientAuthenticationError`.